### ISSUE TITLE
Random containerd crashes

### ISSUE THREAD

**Issue by Eshakk** [docker/docker#32537]
**Description**

We have around 8 to 10 containers on each node in out network. containerd service(we are running on top of coreos) is crashing in some random nodes at random times. Each time containerd goes down, after couple of hours, docker stops responding and whole node goes into zombie state where only restart would fix things.

**Steps to reproduce the issue:**
We have no idea how to reproduce this as it is happening randomly.

**Output of `docker version`:**

```
Client:
 Version:      1.12.6
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   d5236f0
 Built:        Tue Mar 14 21:10:49 2017
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.6
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   d5236f0
 Built:        Tue Mar 14 21:10:49 2017
 OS/Arch:      linux/amd64

```

**Output of `docker info`:**

```
Containers: 9
 Running: 9
 Paused: 0
 Stopped: 0
Images: 14
Server Version: 1.12.6
Storage Driver: overlay
 Backing Filesystem: extfs
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: host bridge null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: seccomp selinux
Kernel Version: 4.9.9-coreos-r1
Operating System: Container Linux by CoreOS 1298.6.0 (Ladybug)
OSType: linux
Architecture: x86_64
CPUs: 70
Total Memory: 58.98 GiB
Name: Server01
ID: PYQ3:JC2K:TNDY:QT6J:5RCU:6V64:TRPC:E44I:5WUB:B3RW:V3VF:GLGI
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Insecure Registries:
 127.0.0.0/8
```

Journal logs:
journal logs of dockerd and containerd service:
sudo journalctl -b -u dockerd -u containerd
[journal_b_docker.txt](https://github.com/docker/docker/files/914575/journal_b_docker.txt)

dockerd with SIGUSR1:
sudo kill -s SIGUSR1 dockerd-pid
[journal_dockerd_sigusr1.txt](https://github.com/docker/docker/files/914572/journal_dockerd_sigusr1.txt)

If there is any specific info you need, I'll happy to help.

**Comment by kalilc** [comment]
I'm having a similiar issue, will be monitoring this.

**Comment by thaJeztah** [comment]
ping @mlaventure

**Comment by mlaventure** [comment]
Unfortunately this is a go runtime bug (cc @cpuguy83 ?)

I also noticed that you are not using our official packages, the commit should be `78d18021e` and they were built with `go 1.6.4`, you may have better luck with those.

**Comment by cpuguy83** [comment]
Yeah, I'm not sure what we can do here, especially since I can't find this panic in 1.7+.

**Comment by mlaventure** [comment]
Closing this since it's out of the `docker/docker` codebase hand.

@Eshakk I would suggest trying the official packages, maybe the issue is less likely to happen with go1.6.4 (if you stick with 1.12.6)

**Comment by Eshakk** [comment]
@mlaventure We cannot change the commit/version since we are using CoreOS and docker is packaged in it. We are having similar crashes in the last few months, not always with the same panic.
@kalilc Do you also have different commit/version than @mlaventure suggested ?

**Comment by kalilc** [comment]
Yes, we were looking into that just minutes ago. We will try to determine the cause of the problem for a while, else we will switch to official package.

**Comment by mlaventure** [comment]
@Eshakk I'd recommend contacting CoreOS then, if the other panic are not related to the go runtime but docker itself, then please open a new issue for each one so we can investigate

**Comment by krmayankk** [comment]
i am seeing this issue with 1.12.6 on Centos. docker-containerd randomly exits. Is this fixed in 1.13.1 or 17.xx  ? Any pointers to why this occurs and which release has this fix is appreciated. @Eshakk @mlaventure @thaJeztah

**Comment by Eshakk** [comment]
@krmayankk I'm not sure whether this is the problem or will fix the issue.

After enabling debug mode for containerd, we have observed one value 'count' under `container-stats-time` to be unusually high. I have not found any documentation regarding this. I've assumed that this would because of running `docker stats` in streaming mode(which we were doing in our setup).

We have fixed this one to run docker stats in no-stream mode and everything is fine since past two months. 

Current versions info where everything is fine:
`docker version`:
```
Client:
 Version:      1.12.6
 API version:  1.24
 Go version:   go1.7.5
 Git commit:   a82d35e
 Built:        Tue Jul 18 23:18:48 2017
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.6
 API version:  1.24
 Go version:   go1.7.5
 Git commit:   a82d35e
 Built:        Tue Jul 18 23:18:48 2017
 OS/Arch:      linux/amd64
```

`docker info`:
```
Containers: 10
 Running: 10
 Paused: 0
 Stopped: 0
Images: 15
Server Version: 1.12.6
Storage Driver: overlay
 Backing Filesystem: extfs
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge null host overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: seccomp selinux
Kernel Version: 4.11.11-coreos
Operating System: Container Linux by CoreOS 1409.7.0 (Ladybug)
OSType: linux
Architecture: x86_64
CPUs: 46
Total Memory: 58.98 GiB
Name: <hostname>
ID: HP6O:34WP:CUSH:WBIG:V2YO:4HGT:EHGB:A2FC:ED3J:2T6Z:KU4Q:2SCN
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Username: <username>
Registry: https://index.docker.io/v1/
Insecure Registries:
 127.0.0.0/8
```
